### DISCUSSION TITLE
Accidentaly deleted files

### DISCUSSION THREAD

**Discussion by freeelancer86** [community/community#78030] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Feature Area

Issues

### Body

I'm having a very similar issue that's described in this thread:
https://github.com/atom/github/issues/1001

In my case I was using git desktop to upload my project folder on Git. My internet was acting up and after several tries that failed to upload, I removed a bunch of files from the folder in order to reduce the workload and try to do it piecemeal. I thought the commit would automatically update the files, so I tried to upload the 4th time, but saw the number of files still being as original. I used "Revert changes in commit" and now all of my files are stuck in the limbo.

Things I've tried to correct my mistake: 
**git fsck --lost-found  ** 
Checking object directories: 100% (256/256), done.
Checking objects: 100% (5742/5742), done.
Verifying commits in commit graph: 100% (1/1), done. 
So, the files are thankfully still there. However, I don't see lost-found folder in my project. Next I tried:
**for blob in $(git fsck --lost-found | awk ‘$2 == "blob" { print $3 }’); do git cat-file -p $blob > $blob.txt; done** 
This returns following message: blob was unexpected at this time.

**git reflog**
07b80f9 (HEAD -> main) HEAD@{0}: revert: Revert "initial commit"
4aa11f7 HEAD@{1}: commit: try 2 of initial commit
bce43ae HEAD@{2}: commit (initial): initial commit

That's it. What do I do now?

### Guidelines

- [X] I have read the above statement and can confirm my post is relevant to the GitHub feature areas [Issues](https://docs.github.com/en/issues/tracking-your-work-with-issues/about-issues) and/or [Projects](https://docs.github.com/en/issues/planning-and-tracking-with-projects/learning-about-projects/about-projects).

**Comment by queenofcorgis** [comment]
Thanks for posting in the GitHub Community, @freeelancer86 ! 

We're happy you're here. You are more likely to get a useful response if you are posting your question in the applicable category, the Projects and Issues category is solely related to conversations around the GitHub products Issues and Projects. I've gone ahead and moved it for you. Good luck!

**Comment by thevindu-w** [comment]
The problem is that you still have the commit with large files in your branch even if you use git revert (or delete the files and commit). Therefore, when you try to push, it will still try to upload the large files.
To fix this, you have to git reset to the last commit before the commit that you added large files, and apply the changes of later commits without adding the large files.
There are some alternative methods mentioned here: [stackoverflow.com/questions/2100907/how-can-i-remove-delete-a-large-file-from-the-commit-history-in-the-git-reposito](https://stackoverflow.com/questions/2100907/how-can-i-remove-delete-a-large-file-from-the-commit-history-in-the-git-reposito)
and
[www.baeldung.com/git-remove-file-commit-history](https://www.baeldung.com/git-remove-file-commit-history)

**Reply by freeelancer86** [reply]
Thank you for your answer and suggestions! 
I am not sure I managed to follow through your explanation, but in case there's miscommunication, the main issue I'm having is trying to get the files back as I didn't back them up before trying to commit to git and reverting the commit "trapped" them inside the .git folder. I have many files, but they are all small in size so I can easily piecemeal them through, but accessing them is currently impossible.

I'd like to think that just reverting the initial reverting of commit would solve my issue, but I'm a newbie and I'm afraid to do anything else that might cause more damage. Here's what I'd like to try, in case I'm not making sense:
![Screenshot 2023-12-06 203815](https://github.com/community/community/assets/81075528/45c323ba-208e-442e-a9f3-38a75afe396b)

**Reply by thevindu-w** [reply]
To get those files, you can checkout to a previous commit. The following command will checkout to the previous commit in your branch. You can run it multiple times to go back one commit at a time.
```
git checkout HEAD^
```

Alternatively, if you know which commit you added those files (or which commit was the last commit before removing them), you can use,
```
git checkout <commit-SHA>
```
to directly go to that commit. But since you have only a few commits (I guess there are only 3 commits), you can use git checkout HEAD^ to check if your files are there in each commit.

**Reply by freeelancer86** [reply]
Thanks again for your reply. I tried using the first command a number of times but it didn't change anything. The second command threw an error when trying to access the SHA. 
I did manage to fix it in the end though. Gathered enough courage and built a new branch using the initial commit and that reverted the changes back. There's probably an easier way to do this by using some of the commands you supplied, but I'm just leaving this here in case another newbie using Git Desktop encounters the same issue.

I'm very grateful for your kindness, effort and attempts to solve this issue for me! Cheers!
